src/Microsoft.ML.Dnn/ImageClassificationTrainer.cs

```diff
@@ -705,10 +705,19 @@ public ImageProcessor(Session session, string jpegDataTensorName, string resizeI
 
             public Tensor ProcessImage(in VBuffer<byte> imageBuffer)
             {
-                var imageTensor = EncodeByteAsString(imageBuffer);
-                var processedTensor = _imagePreprocessingRunner.AddInput(imageTensor, 0).Run()[0];
-                imageTensor.Dispose();
-                return processedTensor;
+                using var imageTensor = EncodeByteAsString(imageBuffer);
```

Comment:
>using var imageTensor = EncodeByteAsString(imageBuffer); [](start = 16, length = 56)

@eric.erhardt@microsoft.com how is the scope of using statement now defined? previously the curly braces explicitly defined it now does this mean it will span until the end of scope in which this variable is created? While I'm 99% sure but when we do a return within a using scope it still ensures the object is disposed before the control leaves the function, right?

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8#using-declarations